redis/src/cluster_async/mod.rs

```diff
@@ -1549,6 +1632,81 @@ impl Connect for MultiplexedConnection {
 }
 
 async fn connect_and_check<C>(
+    addr: &str,
+    params: ClusterParams,
+    socket_addr: Option<SocketAddr>,
+    conn_type: RefreshConnectionType,
+    node: Option<AsyncClusterNode<C>>,
+) -> RedisResult<AsyncClusterNode<C>>
+where
+    C: ConnectionLike + Connect + Send + Sync + 'static + Clone,
+{
+    match conn_type {
+        RefreshConnectionType::OnlyUserConnection => {
+            let (user_conn, ip) = create_user_connection(addr, params.clone(), socket_addr).await?;
+            let management_conn = if let Some(node) = node {
+                node.management_connection.clone()
+            } else {
+                None
+            };
+
+            return Ok(AsyncClusterNode::new(
```

Comment:
minor: looks like all of the `return` statements here could be removed. I wonder why the linter didn't fail on that.